simexpal/instances.py

```diff
@@ -98,8 +99,31 @@ def substitute(p):
 				raise RuntimeError("Unexpected parameter {}".format(p))
 
 			url = expand_at_params(inst_yml['url'], substitute)
-			download_path = partial_path + ext
+			download_path = os.path.join(instances_dir, filename + '.download')
 			download_as_stream(url, download_path)
+
+			if inst_yml.get('format', None):
+				import shutil
+
+				tmp_path = os.path.join(instances_dir, filename + '.tmp')
+				try:
+					shutil.unpack_archive(download_path, extract_dir=instances_dir, format=inst_yml['format'])
```

Comment:
I think we should extract to a temporary subdirectory of instances_dir here. The zip file might contain all kinds of files that clash with other instances.

Reply:
We can use `os.rename` afterwards to move the right file to the instances_dir, and remove the temporary directory.

Reply:
I agree

Reply:
Done.